### ISSUE TITLE
getAttachment*() functionality generally broken on pyral 1.4.2 and Python 3.6

### ISSUE THREAD

**Issue by pgacv2** [RallyTools/RallyRestToolkitForPython#150]
Not sure if this is related to #110 or #138.

According to the [docs](https://pyral.readthedocs.io/en/latest/interface.html):

> **getAttachment(artifact, filename)**
> Given a real artifact instance or the FormattedID of an existing artifact, obtain the attachment named by filename. If there is such an attachment, return an Attachment instance with hydration for Name, Size, ContentType, Content, CreationDate and the User that supplied the attachment. If no such attachment is present, return None
> 
> **getAttachmentNames(artifact)**
> Given a real artifact instance that is hydrated for at least the Attachments attribute, return the names (filenames) of the Attachments associated with the artifact.
> 
> **getAttachments(artifact)**
> Given a real artifact instance, return a list of Attachment records. Each Attachment record will look like a Rally WSAPI Attachment with the additional Content attribute that will contain the decoded AttachmentContent.

```
>>> rally = pyral.Rally(apikey=apikey)
>>> att = rally.getAttachment('US8926', 'Bowling.xlsx')
>>> att
False
```

The attachment is definitely there, and it returns False instead of None. I assigned the US8926 object to `us` and tried again:
```
>>> us
<pyral.entity.HierarchicalRequirement object at 0x0000029E7AD516D8>
>>> us.FormattedID
'US8926'
>>> rally.getAttachmentNames(us)
>>> ['Bowling.xlsx']
>>> filename = rally.getAttachmentNames(us)[0]
>>> att = rally.getAttachment(us, filename)
Traceback (most recent call last):
  File "C:\Program Files\Python36\lib\base64.py", line 517, in _input_type_check
    m = memoryview(s)
TypeError: memoryview: a bytes-like object is required, not 'str'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<pyshell#107>", line 1, in <module>
    att = rally.getAttachment(us, 'Bowling.xlsx')
  File "C:\Program Files\Python36\lib\site-packages\pyral\restapi.py", line 1700, in getAttachment
    att.Content = base64.decodebytes(att_content.Content)  # maybe further txfm to Unicode ?
  File "C:\Program Files\Python36\lib\base64.py", line 552, in decodebytes
    _input_type_check(s)
  File "C:\Program Files\Python36\lib\base64.py", line 520, in _input_type_check
    raise TypeError(msg) from err
TypeError: expected bytes-like object, not str
```

I tried a bytes filename and got None:
```
>>> att = rally.getAttachment(us, b'Bowling.xlsx')
>>> att
>>>
>>> att = rally.getAttachment(us, bytes(filename, encoding='utf-8'))
>>> att
>>>
```

I tried getting all attachments:
```
>>> att = rally.getAttachments(us)
Traceback (most recent call last):
  File "C:\Program Files\Python36\lib\base64.py", line 517, in _input_type_check
    m = memoryview(s)
TypeError: memoryview: a bytes-like object is required, not 'str'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<pyshell#110>", line 1, in <module>
    att = rally.getAttachments(us)
  File "C:\Program Files\Python36\lib\site-packages\pyral\restapi.py", line 1721, in getAttachments
    attachments = [self.getAttachment(artifact, attachment_name) for attachment_name in attachment_names]
  File "C:\Program Files\Python36\lib\site-packages\pyral\restapi.py", line 1721, in <listcomp>
    attachments = [self.getAttachment(artifact, attachment_name) for attachment_name in attachment_names]
  File "C:\Program Files\Python36\lib\site-packages\pyral\restapi.py", line 1700, in getAttachment
    att.Content = base64.decodebytes(att_content.Content)  # maybe further txfm to Unicode ?
  File "C:\Program Files\Python36\lib\base64.py", line 552, in decodebytes
    _input_type_check(s)
  File "C:\Program Files\Python36\lib\base64.py", line 520, in _input_type_check
    raise TypeError(msg) from err
TypeError: expected bytes-like object, not str
```

**Comment by jfthuong** [comment]
I believe it is the same issue as #138.

You can try the fix I have done in _\<Python Installation Directory\>_\Lib\site-packages\pyral\restapi.py: https://github.com/jfthuong/RallyRestToolkitForPython/commit/54f42156905f09318481a83d3ad035acb0c2db3f

**Comment by klehman-rally** [comment]
Fixed in the 1.5.0 release
